core/src/main/scala/kafka/zookeeper/ZooKeeperClient.scala

```diff
@@ -198,6 +207,23 @@ class ZooKeeperClient(connectString: String,
             callback(GetChildrenResponse(Code.get(rc), path, Option(ctx), Option(children).map(_.asScala).getOrElse(Seq.empty),
               stat, responseMetadata(sendTimeMs)))
         }, ctx.orNull)
+      case GetChildrenPaginatedRequest(path, _, ctx) =>
+        CompletableFuture.supplyAsync(new Supplier[GetChildrenPaginatedResponse]() {
+          def get(): GetChildrenPaginatedResponse = {
+            var topicsList: JList[String] = null 
+            var resultCode: Code = Code.OK
+            try {
+              topicsList = zooKeeper.getAllChildrenPaginated(path, shouldWatch(request))
+            } catch {
+              case e: KeeperException =>
```

Comment:
I traced the code. 

For other un-checked exceptions, it would end up `CompletionException`, and the upper level catches those
https://github.com/linkedin/kafka/blob/21cece149e5668dfc4f40ee61fa1741d71d51270/core/src/main/scala/kafka/zookeeper/ZooKeeperClient.scala#L163

Reply:
No, the caller handles all Throwables for all of the APIs. All we need here is the error code from any ZK server errors:

`@throws KeeperException if the server signals an error with a non-zero error code`